quota/etcd/quotaapi/quota_server.go

```diff
@@ -40,15 +73,80 @@ func (s *Server) DeleteConfig(ctx context.Context, req *quotapb.DeleteConfigRequ
 
 // GetConfig implements quotapb.QuotaServer.GetConfig.
 func (s *Server) GetConfig(ctx context.Context, req *quotapb.GetConfigRequest) (*quotapb.Config, error) {
-	return nil, errNotImplemented
+	if req.Name == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "name is required")
+	}
+
+	cfgs, err := s.qs.Configs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	cfg, ok := findByName(req.Name, cfgs)
+	if !ok {
+		return nil, status.Errorf(codes.NotFound, "%q not found", req.Name)
+	}
+	return convertToAPI(cfg), nil
 }
 
-// ListConfig implements quotapb.QuotaServer.ListConfig.
-func (s *Server) ListConfig(ctx context.Context, req *quotapb.ListConfigRequest) (*quotapb.ListConfigResponse, error) {
+// ListConfigs implements quotapb.QuotaServer.ListConfigs.
+func (s *Server) ListConfigs(ctx context.Context, req *quotapb.ListConfigsRequest) (*quotapb.ListConfigsResponse, error) {
 	return nil, errNotImplemented
 }
 
 // UpdateConfig implements quotapb.QuotaServer.UpdateConfig.
 func (s *Server) UpdateConfig(ctx context.Context, req *quotapb.UpdateConfigRequest) (*quotapb.Config, error) {
-	return nil, errNotImplemented
+	hasConfig := req.Config != nil
+	hasMask := req.UpdateMask != nil && len(req.UpdateMask.Paths) > 0
+	switch {
+	case req.Name == "":
+		return nil, status.Errorf(codes.InvalidArgument, "name must be specified")
+	case req.ResetQuota && !hasConfig && !hasMask:
+		// For convenience, reset-only requests are allowed.
+		// Let's "fix" the request so we don't panic later on.
+		req.Config = &quotapb.Config{}
+		req.UpdateMask = &field_mask.FieldMask{}
+	case !hasConfig:
+		return nil, status.Errorf(codes.InvalidArgument, "config must be specified")
+	case !hasMask:
+		return nil, status.Errorf(codes.InvalidArgument, "update_mask must be specified")
+	}
+	if err := validateMask(req.UpdateMask); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid update_mask: %v", err)
+	}
+
+	var notFound bool
+	updated, err := s.qs.UpdateConfigs(ctx, req.ResetQuota, func(cfgs *storagepb.Configs) {
+		cfg, ok := findByName(req.Name, cfgs)
+		if !ok {
+			notFound = true
+			return
+		}
+		applyMask(req.Config, cfg, req.UpdateMask)
+	})
+	switch {
+	case notFound:
+		return nil, status.Errorf(codes.NotFound, "%q not found", req.Name)
+	case err != nil:
+		return nil, err
+	}
+	return replyFromConfigs(req.Name, updated)
+}
+
+func findByName(name string, cfgs *storagepb.Configs) (*storagepb.Config, bool) {
```

Comment:
I'd rather keep the dual return. I think the API is cleaner and an extra boolean is harder to ignore.